server/routes/public.js

```diff
@@ -29,15 +29,48 @@ router.get('/api/login', async (req, res) => {
 });
 
 router.get('/api/create-user', async (req, res) => {
-    const firstName = req.query.firstName;
-    const lastName = req.query.lastName;
-    const type = req.query.type; //note: 'type' is called 'role' in the UI, but refers to either a worker or a student.
-    const studentID = req.query.studentID;
-    const email = req.query.email;
-    const password = req.query.password;
-    const phone = req.query.phone;
-
-    const user = await accountHandler.createUserAccount(firstName, lastName, type, studentID, email, password, phone);
+    // joi verification
+    // Validate appropriate parameters are passed into the create account endpoint.
+    const paramSchema = Joi.object({
+       
+       //The following joi verifications make sure the values inputted adhere to the format requirements
+       //specified in the migrations> create-worker-table and migrations> create-student-table.
+
+        email: Joi.string().email({ minDomainSegments: 2, tlds: { allow: ['com', 'net', 'ca'] } }).max(320).required(),
+        password: Joi.string().min(3).max(40).required(),
+        phone: Joi.string().max(20).phoneNumber(),
+        first_name: Joi.string().min(1).max(20).required(),
+        last_name: Joi.string().min(1).max(20).required(),
+        student_id: Joi.number().positive().integer(),
+        worker_id: Joi.number().positive().integer(),
+        userType: Joi.string().min(1).valid(['student', 'worker']),
+        specialization: Joi.string().valid(['social worker', 'guidance councellor']),//verify options
+        access_token: Joi.string().alphanum().min(3).max(30).required(), 
```

Comment:
Hi Nancy,
I would suggest removing the `access_token` value here as it is no longer being passed into the Joi validation.

Reply:
Hi Amy,
Thank you for suggesting that!
My action: I've removed the access_token from line 49 of server/routes/public.js as you recommended!